packages/eip-7702-internal-rpc-middleware/src/methods/wallet_getAccountUpgradeStatus.test.ts

```diff
@@ -0,0 +1,47 @@
+import type {
+  JsonRpcRequest,
+  PendingJsonRpcResponse,
+  Json,
+} from '@metamask/utils';
+
+import { walletGetAccountUpgradeStatus } from './wallet_getAccountUpgradeStatus';
+import { getAccountUpgradeStatus } from '../hooks/getAccountUpgradeStatus';
+import type { GetAccountUpgradeStatusHooks } from '../types';
+
+jest.mock('../hooks/getAccountUpgradeStatus');
+
+describe('walletGetAccountUpgradeStatus', () => {
+  it('calls getAccountUpgradeStatus hook with correct parameters and sets result', async () => {
+    const req = {} as JsonRpcRequest<Json[]> & { origin: string };
+    const res = { result: null } as PendingJsonRpcResponse;
+    const hooks = {
+      getCurrentChainIdForDomain: jest.fn(),
+      getCode: jest.fn(),
+      getNetworkConfigurationByChainId: jest.fn(),
+      getAccounts: jest.fn(),
+    };
+
+    const mockResult = { account: '0xabc', isUpgraded: true, chainId: 1 };
+    (getAccountUpgradeStatus as jest.Mock).mockResolvedValue(mockResult);
```

Comment:
If we add

``` typescript
const getAccountUpgradeStatusMock = jest.mocked(getAccountUpgradeStatus);
```

to the top of the file, after the imports, then we don't need to assert that this is a `jest.Mock`, we can use `getAccountUpgradeStatusMock` from now on.